ert/storage/_storage.py

```diff
@@ -95,10 +93,12 @@ async def _transmit_blob_record(self, record: ert.data.BlobRecord) -> str:
         return self._uri
 
     async def _load_numerical_record(self) -> ert.data.NumericalRecord:
+        assert self._record_type
```

Comment:
Why is this naked assert needed? I am not really sure how useful naked asserts are in general. I find that they provide at best just half of the information required to understand what is going on when there is an actual problem. 

If you are just trying to add some sanity checks then I would suggest actually checking the type of the record and if there is a mismatch between what you are expecting and the actual record then I would raise an error with a message.  Not sure how useful all of that would be given that a very similar error would have been raised anyways two lines down.  

*Why is this naked assert needed?* I think this question holds true for all the other instances of naked asserts you are adding in this PR

Reply:
ohh... I missed the parameter data type update for `load_record`

Reply:
if assertions are used all over the place due to this Record class, does that imply that the class needs redesigning?
such as:
```python
class Record(_DataElement):
    data: record_data
    record_type: RecordType

    # validator always sets record_type to a not None and pre is True
    # Thus there is no need for Optional
    @validator("record_type", pre=True)
    def record_type_validator(
        cls,
        record_type: Optional[RecordType],
        values: Dict[str, Any],
    ) -> Optional[RecordType]:
        if record_type is None and "data" in values:
            data = values["data"]
            if isinstance(data, list):
                if not data or isinstance(data[0], (int, float)):
                    return RecordType.LIST_FLOAT
            elif isinstance(data, bytes):
                return RecordType.BYTES
            elif isinstance(data, Mapping):
                if not data:
                    return RecordType.MAPPING_STR_FLOAT
                if isinstance(list(data.keys())[0], (int, float)):
                    return RecordType.MAPPING_INT_FLOAT
                if isinstance(list(data.keys())[0], str):
                    return RecordType.MAPPING_STR_FLOAT
        return record_type

    def get_instance(self) -> "Record":
        if self.record_type is not None:
            if self.record_type == RecordType.BYTES:
                return BlobRecord(data=self.data)
        return NumericalRecord(data=self.data)
```

Reply:
The example above will not work, but I still stand that the assertions are an antipattern

Reply:
This will not work.... I think if you set it up like 
```python 
class Record(_DataElement): 
   data: record_data
   record_type: RecordType
```

It means the record_type filed is required when you are making a record and if you have to provide that there is no more point in the validation